src/main/java/at/ac/tuwien/kr/alpha/grounder/atoms/EnumerationAtom.java

```diff
@@ -52,14 +52,16 @@ private Integer getEnumerationIndex(Term identifier, Term enumerationTerm) {
 
 	}
 
-	public void addEnumerationToSubstitution(Substitution substitution) {
+	public Substitution addEnumerationIndexToSubstitution(Substitution substitution) {
 		Term idTerm = getTerms().get(0).substitute(substitution);
 		Term enumerationTerm  = getTerms().get(1).substitute(substitution);
 		if (!enumerationTerm.isGround()) {
 			throw new RuntimeException("Enumeration term is not ground after substitution. Should not happen.");
 		}
 		Integer enumerationIndex = getEnumerationIndex(idTerm, enumerationTerm);
-		substitution.put((VariableTerm) getTerms().get(2), ConstantTerm.getInstance(enumerationIndex));
+		Substitution retVal = new Substitution(substitution);
+		retVal.put((VariableTerm) getTerms().get(2), ConstantTerm.getInstance(enumerationIndex));
+		return retVal;
```

Comment:
No, that's actually meant to be "real" code - The change I made here is that `addEnumerationIndexToSubstitution` returns a new substitution with the enum index added instead of doing this in place on the given substitution. I wanted the method to be consistent with similar methods on `FixedInterPretationLiteral`s where - at least in most cases - new substitutions are returned rather than working directly on the passed-in ones.

Reply:
Oh right, I didn't catch that.